src/utils/atomWithSuspense.ts

```diff
@@ -0,0 +1,26 @@
+import { atom } from 'jotai'
+import type { SetStateAction } from 'jotai'
+
+export function atomWithSuspense<Value>() {
+  const SUSPENSE = Symbol()
+  let ready: (value: Value) => void
```

Comment:
Implementation-wise, this has an issue. This doesn't work with multi provider use cases. We can't have a module state outside atoms.

Reply:
Oh, really, then I don't think I've delved enough into internal implementation.